pytools/tag.py

```diff
@@ -321,6 +323,44 @@ def tags_of_type(self, tag_t: Type[TagT]) -> FrozenSet[TagT]:
                          for tag in self.tags
                          if isinstance(tag, tag_t)})
 
+    @memoize_method
+    def tags_not_of_type(self, tag_t: Type[TagT]) -> FrozenSet[Tag]:
+        """
+        Returns *self*'s tags that are not of type *tag_t*.
+        """
+        return frozenset({tag
+                         for tag in self.tags
+                         if not isinstance(tag, tag_t)})
+
+    def __eq__(self, __o: object) -> bool:
```

Comment:
Double-underscore identifiers are reserved in Python, AFAIK.

Reply:
Fixed in [b781d63](https://github.com/inducer/pytools/pull/134/commits/b781d6302ca0294fb1534fe2177fa5eef95111ce).\

Interestingly, `def __eq__(self, __o: object) -> bool` is the default signature of `__eq__` that my editor (VS Code) suggests.